Source/AwsCommonRuntimeKit/mqtt/Mqtt5Packets.swift

```diff
@@ -179,63 +179,105 @@ public class PublishPacket: CStruct {
+                    raw_publish_view.message_expiry_interval_seconds = messageExpiryIntervalPointer
+                    raw_publish_view.topic_alias = topicAliasPointer
+                        
                     return withOptionalArrayRawPointer(of: subscriptionIdentifiers) { subscriptionPointer in
 
-                        if let _subscriptionPointer = subscriptionPointer {
-                            raw_publish_view.subscription_identifiers = _subscriptionPointer
+                        if let subscriptionPointer = subscriptionPointer {
+                            raw_publish_view.subscription_identifiers = subscriptionPointer
                             raw_publish_view.subscription_identifier_count = subscriptionIdentifiers!.count
                         }
 
                         return withOptionalUserPropertyArray(
                             of: userProperties) { userPropertyPointer in
-                            if let _userPropertyPointer = userPropertyPointer {
+                            if let userPropertyPointer = userPropertyPointer {
                                 raw_publish_view.user_property_count = userProperties!.count
                                 raw_publish_view.user_properties =
-                                    UnsafePointer<aws_mqtt5_user_property>(_userPropertyPointer)
+                                    UnsafePointer<aws_mqtt5_user_property>(userPropertyPointer)
                             }
-                            return withOptionalByteCursorPointerFromString(
+                            return withOptionalByteCursorPointerFromStrings(
                             responseTopic,
-                            correlationData,
-                            contentType) { cResponseTopic, cCorrelationData, cContentType in
+                            contentType) { cResponseTopic, cContentType in
                                 raw_publish_view.content_type = cContentType
-                                raw_publish_view.correlation_data = cCorrelationData
                                 raw_publish_view.response_topic = cResponseTopic
 
-                                return body(raw_publish_view)
+                                return withOptionalByteCursorPointerFromData(to: self.correlationData) {  cCorrelationData in
+                                    raw_publish_view.correlation_data = cCorrelationData
+                                    return body(raw_publish_view)
+                                }
                             }
                         }
                     }
                 }
             }
         }
     }
+
+    static func convertFromNative(_ from: UnsafePointer<aws_mqtt5_packet_publish_view>?) -> PublishPacket? {
+        guard let from = from else {
+            return nil
+        }
+        let publishView = from.pointee
+
+        let payload: Data = Data(bytes: publishView.payload.ptr, count: publishView.payload.len)
```

Comment:
nit / debatable: If the type is really obvious, I think it makes it cleaner to just omit it or at least keep it consistent.
Same comment for others.
```suggestion
        let payload = Data(bytes: publishView.payload.ptr, count: publishView.payload.len)
```